discovery-provider/plugins/notifications/src/processNotifications/mappers/usdcPurchaseSeller.ts

```diff
@@ -0,0 +1,171 @@
+import { Knex } from 'knex'
+import { NotificationRow } from '../../types/dn'
+import {
+  USDCPurchaseSellerNotification,
+  AppEmailNotification
+} from '../../types/notifications'
+import { BaseNotification } from './base'
+import { sendPushNotification } from '../../sns'
+import { ResourceIds, Resources } from '../../email/notifications/renderEmail'
+import { sendNotificationEmail } from '../../email/notifications/sendEmail'
+import {
+  buildUserNotificationSettings,
+  Device
+} from './userNotificationSettings'
+import { disableDeviceArns } from '../../utils/disableArnEndpoint'
+import { capitalize } from 'lodash'
+import { sendBrowserNotification } from '../../web'
+import { EntityType } from '../../email/notifications/types'
+
+type USDCPurchaseSellerRow = Omit<NotificationRow, 'data'> & {
+  data: USDCPurchaseSellerNotification
+}
+export class USDCPurchaseSeller extends BaseNotification<USDCPurchaseSellerRow> {
+  notificationReceiverUserId: number
+  buyerUserId: number
+  amount: number
+  content_id: number
+
+  constructor(
+    dnDB: Knex,
+    identityDB: Knex,
+    notification: USDCPurchaseSellerRow
+  ) {
+    super(dnDB, identityDB, notification)
+    const userIds: number[] = this.notification.user_ids!
+    this.amount = this.notification.data.amount
+    this.buyerUserId = this.notification.data.buyer_user_id
+    this.notificationReceiverUserId = this.notification.data.seller_user_id
+    this.content_id = this.notification.data.content_id
+  }
+
+  async processNotification({
+    isLiveEmailEnabled,
+    isBrowserPushEnabled
+  }: {
+    isLiveEmailEnabled: boolean
+    isBrowserPushEnabled: boolean
+  }) {
+    const users = await this.getUsersBasicInfo([
+      this.notificationReceiverUserId,
+      this.buyerUserId
+    ])
+    if (users?.[this.notificationReceiverUserId]?.isDeactivated) {
+      return
+    }
+    // Get the user's notification setting from identity service
+    const userNotificationSettings = await buildUserNotificationSettings(
+      this.identityDB,
+      [this.notificationReceiverUserId, this.buyerUserId]
+    )
+
+    const tracks = await this.fetchEntities([this.content_id], EntityType.Track)
+    let purchasedTrackName
+    if ('title' in tracks[this.content_id]) {
+      purchasedTrackName = (tracks[this.content_id] as { title: string }).title
+    }
+    const buyerUsername = users[this.buyerUserId]?.name
+    // TODO: Convert to usdc
+    const price = this.amount
+
+    const title = 'Track Sold'
+    const body = `Congrats, ${capitalize(
```

Comment:
Not sure what the policy is in platform land, but it might be easier to move string literals and simple functions that generate them up to constants at the top of the file.

Reply:
no policy really, but the convention has been to keep them here. i do like that they're at the top of the file in client land though so will give a shot